### PR TITLE
Dev/dmitrypa sdl zero pointers

### PR DESCRIPTION
**Pull request by dmitry741**
Zero pointers according to SDL requirements.

### CODE REVIEW

File: algorithms/kernel/brownboost/brownboost_train_impl.i [hunk 1]
```diff
@@ -320,7 +327,11 @@ services::Status BrownBoostTrainKernelNew<method, algorithmFPType, cpu>::compute
     }
     }
 
-    if (alpha) { daal::services::daal_free(alpha); }
+    if (alpha)
```
Comment (SmirnovEgorRu) [inline algorithms/kernel/brownboost/brownboost_train_impl.i]: FYI (no changes are required):
free() checks ptr for nullptr and does nothing if it is null ptr.

**Comment by SmirnovEgorRu** [comment]
make test

**Comment by SmirnovEgorRu** [comment]
I see some failed "clear code" checks.

**Comment by ShvetsKS** [comment]
please fix clear code problems in files:
algorithms/kernel/brownboost/brownboost_train_impl.i
algorithms/kernel/dtrees/gbt/gbt_train_dense_default_impl.i
algorithms/kernel/naivebayes/naivebayes_train_container.h
externals/service_stat_mkl.h

**Comment by SmirnovEgorRu** [comment]
make test

**Pull request merged**